### PR TITLE
Tests: improve coverage of Network

### PR DESCRIPTION
**Pull request by erlend-aasland**

### CODE REVIEW

**Comment by erlend-aasland** [comment]
I'm not sure if I'm satisfied with this yet; I'll have another look tomorrow before marking it as ready for review.

File: test/test_network.py [hunk 1]
```diff
@@ -9,28 +10,84 @@
...
+        self.assertEqual(node.id, 5)
+        self.assertIsInstance(node, canopen.LocalNode)
+
+        # Verify that we've got the correct number of nodes.
+        self.assertEqual(len(self.network), 4)
+
+    def test_network_add_node_upload_eds(self):
+        # Will err because we're not connected to a real network.
+        with self.assertLogs(level=logging.ERROR):
+            self.network.add_node(2, SAMPLE_EDS, upload_eds=True)
```
Comment (erlend-aasland) [inline test/test_network.py:51]: This test can be enhanced for example by mocking `canopen.sdo.SdoClient.open` to return a pre-filled StringIO buffer.

**Review by acolomb — APPROVED** [review]
I haven't checked every detail, but the code looks sensible, clean and appears to test the relevant bits.  Fine with me, let's take a closer look when something should ever fail.

**Comment by erlend-aasland** [comment]
Thank you for the review!

**Pull request merged**